bundles/org.openhab.core.model.yaml/src/main/java/org/openhab/core/model/yaml/YamlElement.java

```diff
@@ -12,28 +12,54 @@
  */
 package org.openhab.core.model.yaml;
 
-import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.NonNull;
+import org.openhab.core.model.yaml.internal.YamlModelRepositoryImpl;
 
 /**
- * The {@link YamlElement} interface offers an identifier and a check validity method
- * to any element defined in a YAML configuration file.
+ * The {@link YamlElement} interface must be implemented by any classes that need to be handled by the
+ * {@link YamlModelRepositoryImpl}.
+ * <p />
+ * Implementations
+ * <ul>
+ * <li>MUST have a default constructor to allow deserialization with Jackson</li>
+ * <li>MUST provide {@code equals(Object other)} and {@code hashcode()} methods</li>
+ * <li>MUST be annotated with {@link YamlElementName} containing the element name</li>
+ * <li>SHOULD implement a proper {@code toString()} method</li>
+ * </ul>
  *
  * @author Laurent Garnier - Initial contribution
+ * @author Jan N. Klug - Refactoring and improvments to JavaDoc
  */
-@NonNullByDefault
```

Comment:
Why was this removed?

Reply:
This essentially a DTO. Missing elements in the data are de-serialized as `null`.

If we decide to make the DTO classes `@NonNullByDefault` we would need to either annotate all fields with `@Nullable` (which is the similar to not annotating the class at all) or provide default values (which doesn't make sense in most cases).